src/plugins/dashboard/common/embeddable/dashboard_control_group.ts

```diff
@@ -7,57 +7,73 @@
  */
 
 import { SerializableRecord } from '@kbn/utility-types';
-import { ControlGroupInput } from '../../../controls/common';
-import { ControlStyle } from '../../../controls/common/types';
+import { ControlGroupInput, getDefaultControlGroupInput } from '../../../controls/common';
 import { RawControlGroupAttributes } from '../types';
 
+export const getDefaultDashboardControlGroupInput = getDefaultControlGroupInput;
+
 export const controlGroupInputToRawAttributes = (
   controlGroupInput: Omit<ControlGroupInput, 'id'>
-): Omit<RawControlGroupAttributes, 'id'> => {
+): RawControlGroupAttributes => {
   return {
     controlStyle: controlGroupInput.controlStyle,
+    chainingSystem: controlGroupInput.chainingSystem,
     panelsJSON: JSON.stringify(controlGroupInput.panels),
+    ignoreParentSettingsJSON: JSON.stringify(controlGroupInput.ignoreParentSettings),
   };
 };
 
-export const getDefaultDashboardControlGroupInput = () => ({
-  controlStyle: 'oneLine' as ControlGroupInput['controlStyle'],
-  panels: {},
-});
-
 export const rawAttributesToControlGroupInput = (
-  rawControlGroupAttributes: Omit<RawControlGroupAttributes, 'id'>
+  rawControlGroupAttributes: RawControlGroupAttributes
 ): Omit<ControlGroupInput, 'id'> | undefined => {
-  const defaultControlGroupInput = getDefaultDashboardControlGroupInput();
+  const defaultControlGroupInput = getDefaultControlGroupInput();
+  const { chainingSystem, controlStyle, ignoreParentSettingsJSON, panelsJSON } =
+    rawControlGroupAttributes;
+  const panels =
+    panelsJSON && typeof rawControlGroupAttributes?.panelsJSON === 'string'
+      ? JSON.parse(rawControlGroupAttributes?.panelsJSON)
```

Comment:
Should we try/catch all of these JSON.parses? Just trying to think of places that have bitten us in the past especially on migrations where customers randomly get their data in a real bad way.

Reply:
Great call, will do that! We need to be as defensive as possible in these.